crates/router/src/core/connector_validation.rs

```diff
@@ -0,0 +1,581 @@
+use api_models::enums as api_enums;
+use common_utils::pii;
+use error_stack::ResultExt;
+use external_services::http_client::client;
+use masking::PeekInterface;
+use pm_auth::connector::plaid::transformers::PlaidAuthType;
+
+use crate::{
+    core::errors::{self},
+    types,
+    types::transformers::ForeignTryFrom,
+};
+
+pub struct ConnectorAuthTypeAndMetadataValidation<'a> {
+    pub connector_name: &'a api_models::enums::Connector,
+    pub auth_type: &'a types::ConnectorAuthType,
+    pub connector_meta_data: &'a Option<pii::SecretSerdeValue>,
+}
+
+impl ConnectorAuthTypeAndMetadataValidation<'_> {
+    pub fn validate_auth_and_metadata_type(
+        &self,
+    ) -> Result<(), error_stack::Report<errors::ApiErrorResponse>> {
+        let connector_auth_type_validation = ConnectorAuthTypeValidation {
+            auth_type: self.auth_type,
+        };
+        connector_auth_type_validation.validate_connector_auth_type()?;
+        self.validate_auth_and_metadata_type_with_connector()
+            .map_err(|err| match *err.current_context() {
+                errors::ConnectorError::InvalidConnectorName => {
+                    err.change_context(errors::ApiErrorResponse::InvalidRequestData {
+                        message: "The connector name is invalid".to_string(),
+                    })
+                }
+                errors::ConnectorError::InvalidConnectorConfig { config: field_name } => err
+                    .change_context(errors::ApiErrorResponse::InvalidRequestData {
+                        message: format!("The {} is invalid", field_name),
```

Comment:
Please inline format args, clippy throws warning about these in Rust 1.88.

Reference: #8498.

```suggestion
                        message: format!("The {field_name} is invalid"),
```

You can also update to latest stable toolchain, merge main to your branch and address any such warnings that may remain.